src/reg.c

```diff
@@ -260,6 +268,30 @@ int reg_debug(struct re_printf *pf, const struct reg *reg)
 }
 
 
+/**
+ * Print the registration information in JSON
+ *
+ * @param pf  Print function
+ * @param ua  Registration object
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int reg_json_api(struct odict *od, const struct reg *reg)
+{
+	int err = 0;
+
+	if (!reg)
+		return 0;
+
+	err |= odict_entry_add(od, "id", ODICT_INT, reg->id);
+	err |= odict_entry_add(od, "scode", ODICT_INT, reg->scode);
```

Comment:
odict_entry_add is using va_list, so for type `ODICT_INT` you need to cast
the value to `int64_t`

(both id and scode)